Client/Frontend/Browser/BackForwardListAnimator.swift

```diff
@@ -0,0 +1,80 @@
+    
+    func animateTransition(transitionContext: UIViewControllerContextTransitioning) {
+        let screens = (from: transitionContext.viewControllerForKey(UITransitionContextFromViewControllerKey)!, to: transitionContext.viewControllerForKey(UITransitionContextToViewControllerKey)!)
+        
+        guard let backForwardViewController = !self.presenting ? screens.from as? BackForwardListViewController : screens.to as? BackForwardListViewController else {
+            return
+        }
+        
+        var bottomViewController = !self.presenting ? screens.to as UIViewController : screens.from as UIViewController
+        
+        if let navController = bottomViewController as? UINavigationController {
+            bottomViewController = navController.viewControllers.last ?? bottomViewController
+        }
+        
+        if let browserViewController = bottomViewController as? BrowserViewController {
+            animateWithBackForward(backForwardViewController, browserViewController: browserViewController, transitionContext: transitionContext)
+        }
+    }
+    
+    func transitionDuration(transitionContext: UIViewControllerContextTransitioning?) -> NSTimeInterval {
+        return animationDuration
+    }
+}
+
+extension BackForwardListAnimator: UIViewControllerTransitioningDelegate {
+    func animationControllerForPresentedController(presented: UIViewController, presentingController presenting: UIViewController, sourceController source: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+        self.presenting = true
+        return self
+    }
+    
+    func animationControllerForDismissedController(dismissed: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+        self.presenting = false
+        return self
+    }
+}
+
+extension BackForwardListAnimator {
+    private func animateWithBackForward(backForward: BackForwardListViewController, browserViewController bvc: BrowserViewController, transitionContext: UIViewControllerContextTransitioning) {
+        let containerView = transitionContext.containerView();
+        
+        if(presenting == true) {
+            backForward.view.frame = bvc.parentViewController!.view.frame
+            backForward.view.alpha = 0;
+            backForward.view.layoutIfNeeded()
+            
+            UIView.animateWithDuration(transitionDuration(transitionContext), delay: 0, usingSpringWithDamping: 0.8, initialSpringVelocity: 0.3, options: [], animations: { () -> Void in
+                backForward.view.alpha = 1;
+                containerView!.addSubview(backForward.view)
+                backForward.tableView.snp_updateConstraints { make in
+                    make.height.equalTo(backForward.tableHeight)
+                }
+                backForward.view.layoutIfNeeded()
+                }, completion: { (completed) -> Void in
+                    transitionContext.completeTransition(completed)
+            })
+            
```

Comment:
Nit: Replace any uses of `!` with a `guard` at the top to prevent potential NPEs.